pkg/operator/controllers/gameserverbuild_controller.go

```diff
@@ -37,32 +37,40 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
-// We have observed cases in which we'll create more than one GameServer for a GameServerBuild
-// This is because on the first reconcile we'll see that we have 0 GameServers and we'll create some
+// We have observed cases in which the controller creates more GameServers than necessary for a GameServerBuild
+// This is because e.g. on the first reconcile call, the controller sees that we have 0 GameServers and it starts to create some
 // On the subsequent reconcile, the cache might have not been updated yet, so we'll still see 0 GameServers (or less than asked) and create more,
 // so eventually we'll end up with more GameServers than requested
-// The code will handle and delete the extra GameServers eventually, but it's a waste of resources unfortunately.
-// The solution is to create a synchonized map (since it will be accessed by multiple reconciliations - one for each GameServerBuild)
-// In this map, the key is GameServerBuild.Name whereas the value is map[string]interface{} and contains the GameServer.Name for all the GameServers under creation
-// We use map[string]interface{} instead a []string to facilitate constant time lookups for GameServer names.
-// On every reconcile loop, we check if all the GameServers for this GameServerBuild are present in cache)
+// The controller code will eventually delete the extra GameServers, but we can improve this process.
+// The solution is to create a synchonized map to track which objects were just created
+// (synchronized since it might be accessed by multiple reconciliation goroutines - one for each GameServerBuild)
+// In this map, the key is GameServerBuild.Name
+// The value is a struct with map[string]interface{} and a mutex
+// The map acts like a set which contains the GameServer.Name for all the GameServers under creation
```

Comment:
space after period

Reply:
it's a property :D let me make it more clear!